mmv1/third_party/terraform/utils/framework_location.go

```diff
@@ -0,0 +1,99 @@
+package google
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type LocationDescriber interface {
+	getLocationDescription(providerConfig *frameworkProvider) LocationDescription
```

Comment:
It could be that the function received the default region and zone values as arguments instead of the whole provider config.

```
getLocationDescription(providerRegion, providerZone types.String) LocationDescription {
	// do stuff
}
```